sklearn/gaussian_process/_gpc.py

```diff
@@ -503,7 +503,8 @@ class GaussianProcessClassifier(ClassifierMixin, BaseEstimator):
     kernel : kernel instance, default=None
         The kernel specifying the covariance function of the GP. If None is
         passed, the kernel "1.0 * RBF(1.0)" is used as default. Note that
-        the kernel's hyperparameters are optimized during fitting.
+        the kernel's hyperparameters are optimized during fitting. Also kernel
+        cannot be a CompoundKernel as it is a private API.
```

Comment:
Strictly speaking CompoundKernel is public API since it has it's own page in the [public API docs](https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.kernels.CompoundKernel.html#sklearn.gaussian_process.kernels.CompoundKernel)

```suggestion
        cannot be a `CompoundKernel`.
```

Reply:
sure, I will follow your suggestion